pkg/gadgets/socket-collector/gadget.go

```diff
@@ -70,11 +70,7 @@ func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
 	selector := gadgets.ContainerSelectorFromContainerFilter(trace.Spec.Filter)
 	filteredContainers := t.resolver.GetContainersBySelector(selector)
 	if len(filteredContainers) == 0 {
-		log.Warningf("No container matches the requested filter: %+v", *trace.Spec.Filter)
-
-		trace.Status.OperationError = ""
-		trace.Status.Output = ""
-		trace.Status.State = "Completed"
+		trace.Status.OperationError = gadgets.ErrNoMatch.Error()
```

Comment:
The fact that it is just a tracing is making it difficult to automate parsing. If the string `no container matches the requested filter` changes between IG versions, a Headlamp plugin would have difficulties to know if this is a fatal error or not.

Maybe we need to add a field `OperatorWarning` in the CRD?

Reply:
It would make sense. If we do so, we should just print the `Status.OperationError` anytime it is present while the `Status.OperationWarning` only if `successNodeCount != 0`. I am just wondering how to manage the CRD versioning but let's first see what other folks think about this idea.

cc @mauriciovasquezbernal

Reply:
> difficulties to know if this is a fatal error or not.

I see two options. 

1. Define a list of standard errors that are stable over time.  By doing so users will understand if the error is `ErrNoMatch` or something else. 
2. Separate errors from warnings. ErrNoMatch would be a warning then and the upper layer should check if all nodes fail with ErrNoMatch to raise an error to the user.

Reply:
In my opinion, I would prefer to add the `OperationWarning` into the CRD. This will simplify the logic on the client-side because it removes the string comparison and reduces the logic to:
1. Print every `OperationError` reported by a node.
2. Print `OperationWarning` only if any node generated a valid output.

In addition, as Mauro suggested, we can improve the UX experience by avoiding printing the same error or warning for every node if they are all the same. It will prevent this kind of output:

```
$ kubectl gadget socket-collector -n demo
WARN[0002] Error getting traces from node "my-node-1": no container matches the requested filter  args="[]" command="kubectl-gadget socket-collector"
WARN[0002] Error getting traces from node "my-node-2": no container matches the requested filter  args="[]" command="kubectl-gadget socket-collector"
WARN[0002] Error getting traces from node "my-node-3": no container matches the requested filter  args="[]" command="kubectl-gadget socket-collector"
...
WARN[0002] Error getting traces from node "my-node-N": no container matches the requested filter  args="[]" command="kubectl-gadget socket-collector"
FATA[0002] Error getting traces from all nodes           args="[]" command="kubectl-gadget socket-collector"
```

If you all agree, I will work on it.

Reply:
PR updated, PTAL.